integration_tests/__tests__/test_path_pattern_reporter_message.test.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+'use strict';
+
+import runJest from '../runJest';
+import {cleanup, writeFiles} from '../utils';
+import os from 'os';
+import path from 'path';
+
+const skipOnWindows = require('skipOnWindows');
+const DIR = path.resolve(os.tmpdir(), 'jest_path_pattern_reporter_message');
+
+skipOnWindows.suite();
+
+beforeEach(() => cleanup(DIR));
+afterEach(() => cleanup(DIR));
+
+test('prints a message with path pattern at the end', () => {
+  writeFiles(DIR, {
+    '.watchmanconfig': '',
+    '__tests__/a.test.js': `test('a', () => {});`,
+    '__tests__/b.test.js': `test('b', () => {});`,
+    'package.json': '{}',
+  });
+  let stderr;
+
+  ({stderr} = runJest(DIR, ['a']));
+  expect(stderr).toMatch('Ran all test suites matching "a"');
```

Comment:
there is a lot of complexity around this case in the codebase.
We create a `testPathPattern` string and pass it all the way through the entire test flow to `SummaryReporter` just so that we can print `"a"` instead of `/a/`, even though if we pass multiple parrterns we still print `/a|b` at the end.

example:
```sh
 $> ./jest matchers.test
...
Ran all test suites matching "matchers.test".
```

but:
```sh
$> ./jest matchers.test matchers1.test
Ran all test suites matching /matchers.test|matchers1.test/.
```

this complexity can be avoided if we always print the pattern (and not a string). This way we can just put `testPathPattern` into `globalConfig` and remove it from everywhere else (run_jest.js, TestSelectionConfig, ReporterOptions, formatTestPathPattern, argv and a few other places).

if we do this, the only change to this test will be this like, and it'll look like this:
```js
({stderr} = runJest(DIR, ['a']));
expect(stderr).toMatch('Ran all test suites matching /a/');
```